dashboards/ebs/ebs_edges.sp

```diff
@@ -0,0 +1,52 @@
+edge "ebs_snapshot_to_ec2_ami" {
+  title = "ami"
+
+  sql = <<-EOQ
+    select
+      images.image_id as to_id,
+      s.arn as from_id
+    from
+      aws_ec2_ami as images,
+      jsonb_array_elements(images.block_device_mappings) as bdm,
+      aws_ebs_snapshot as s
+    where
+      bdm -> 'Ebs' is not null
+      and bdm -> 'Ebs' ->> 'SnapshotId' = s.snapshot_id
+      and s.arn = any($1);
+  EOQ
+
+  param "ebs_snapshot_arns" {}
+}
+
+edge "ebs_volume_to_ebs_snapshot" {
+  title = "snapshot"
+
+  sql = <<-EOQ
+    select
+      v.arn as from_id,
+      s.arn as to_id
+    from
+      aws_ebs_snapshot as s,
+      aws_ebs_volume as v
+    where
+      s.snapshot_id = v.snapshot_id
+      and v.arn = any($1);
+  EOQ
+
+  param "ebs_volume_arns" {}
+}
+
+edge "ebs_snapshot_to_kms_key" {
+  title = "encrypted with"
+
+  sql = <<-EOQ
+    select
+      s.arn as from_id,
+      k.arn as to_id
+    from
+      aws_ebs_snapshot as s
+      left join aws_kms_key as k on s.kms_key_id = k.arn and s.arn = any($1);
```

Comment:
Join is not required here as the value of kms_key_id present in aws_ebs_snapshot is key arn.